File: src/main/kotlin/org/ahant/inputvalidator/util/CommonUtil.java
package org.ahant.inputvalidator.util;

import com.google.common.base.Strings;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by ahant on 7/23/2016.
 */
public class CommonUtil {
    private static Logger logger;

    private CommonUtil() {
        throw new IllegalAccessError("Utility class");
    }

    public static boolean isBlank(String input) {
        return Strings.isNullOrEmpty(input);
    }

    public static boolean isNotBlank(String input) {
        return !isBlank(input);
    }

    public static void log(Class<?> clazz, LogSeverity severity, String msg) {
        logger = LoggerFactory.getLogger(clazz);
        switch (severity) {
            case DEBUG:
                logger.debug(msg);
                break;
            case INFO:
                logger.info(msg);
                break;
            case WARN:
                logger.warn(msg);
                break;
            case ERROR:
                logger.error(msg);
                break;
        }
    }

    public static void log(Class<?> clazz, String msg) {
        log(clazz, LogSeverity.DEBUG, msg);
    }

    public enum LogSeverity {
        DEBUG, INFO, WARN, ERROR;
    }
}

File: src/main/kotlin/org/ahant/validator/util/CommonUtil.kt
package org.ahant.validator.util

import com.google.common.base.Strings
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import kotlin.reflect.KClass

/**
 * Created by ahant on 7/23/2016.
 */
class CommonUtil private constructor() {

    init {
        throw IllegalAccessError("Utility class")
    }

    enum class LogSeverity {
        DEBUG, INFO, WARN, ERROR
    }

    companion object {
        private var logger: Logger? = null

        fun isBlank(input: String): Boolean {
            return Strings.isNullOrEmpty(input)
        }

        fun isNotBlank(input: String): Boolean {
            return !isBlank(input)
        }

        fun log(clazz: KClass<*>, severity: LogSeverity, msg: String) {
            logger = LoggerFactory.getLogger(clazz.javaObjectType)
            when (severity) {
                CommonUtil.LogSeverity.DEBUG -> logger!!.debug(msg)
                CommonUtil.LogSeverity.INFO -> logger!!.info(msg)
                CommonUtil.LogSeverity.WARN -> logger!!.warn(msg)
                CommonUtil.LogSeverity.ERROR -> logger!!.error(msg)
            }
        }

        fun log(clazz: KClass<*>, msg: String) {
            log(clazz, LogSeverity.DEBUG, msg)
        }
    }
}
